openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ReturnTypeGenerator.java

```diff
@@ -87,13 +86,29 @@
  */
 public class ReturnTypeGenerator {
 
+    private final BallerinaTypesGenerator ballerinaSchemaGenerator;
+    private final String pathRecord;
+    private static int countForRecord = 0;
+
     private final Map<String, TypeDefinitionNode> typeInclusionRecords = new HashMap<>();
 
     public Map<String, TypeDefinitionNode> getTypeInclusionRecords() {
-
         return this.typeInclusionRecords;
     }
 
+    public static void setCount(int count) {
```

Comment:
Since there is no outside usage of this method cant we use only the non static method to set the count while maintaining a non-static private count variable.

Reply:
This ReturnTypeGenerator creates an instance each http method under one Path. Therefore we need all the newly generated record counts for generating a new record name for one http method, then we need to make static this filed there.

Reply:
@lnash94 One more concern.. does this counter need to be concurrent-safe? (It depends on how the return type generator instances are created and run.)

Reply:
Return-type instances are created and run in a sequence way. Here we need some members' previous value to access.  Therefore had to use class members to achieve it. so hope the counter does not need to be concurrent-safe since this happenes sequence way.